packages/next/src/server/app-render/index.tsx

```diff
@@ -720,8 +721,24 @@ export async function renderToHTMLOrFlight(
 
       const isClientComponent = isClientReference(layoutOrPageMod)
 
+      // If it's a not found route, and we don't have any matched parallel
+      // routes, we try to render the not found component if it exists.
+      let notFoundComponent = {}
+      if (asNotFound && !parallelRouteMap.length && NotFound) {
```

Comment:
do we need to check && NotFound ? I thought it's always defined since we also have a default one?